Calendar/Decorators/ScheduleDecorator.m

```diff
@@ -0,0 +1,88 @@
+//
+//  ScheduleDecorator.m
+//  Calendar
+//
+//  Created by Angelina Zhang on 7/9/22.
+//
+
+#import "ScheduleDecorator.h"
+#import "ScheduleHour.h"
+#import "ScheduleEventView.h"
+
+@implementation ScheduleDecorator
+
+- (void)decorateBaseScheduleWithDate:(NSDate *)date
+                         contentView:(ScheduleScrollView *)view{
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setLocalizedDateFormatFromTemplate:@"EEEE, MMM d, yyyy"];
+    view.dateLabel.text = [dateFormatter stringFromDate:date];
+    
+    int height = view.scrollView.frame.size.height / 24;
+    for (int index = 0; index < 24; index++) {
+        ScheduleHour *hourView = [ScheduleHour new];
+        hourView.timeLabel.text = [self getTimeStringWithIndex:index];
+        hourView.translatesAutoresizingMaskIntoConstraints = false;
+        [view.scrollView addSubview:hourView];
+        [[hourView.widthAnchor constraintEqualToConstant:view.scrollView.frame.size.width] setActive:true];
+        [[hourView.heightAnchor constraintEqualToConstant:height] setActive:true];
+        [[hourView.topAnchor constraintEqualToAnchor:view.scrollView.topAnchor
+                                            constant:height * index] setActive:true];
+        [[hourView.centerXAnchor constraintEqualToAnchor:view.scrollView.centerXAnchor] setActive:true];
+    }
+}
+
+- (void)addEvents:(NSArray<Event *> *)newEvents
+      contentView:(ScheduleScrollView *)view {
+    for (Event *event in newEvents) {
+        [self addEvent:event contentView:view];
+    }
+}
+
+- (void)addEvent:(Event *)newEvent
+     contentView:(ScheduleScrollView *)view {
+    NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
+    [calendar setTimeZone:[NSTimeZone systemTimeZone]];
```

Comment:
Here's a fun question. What will happen if an event was created in a different time zone?